src/Storages/HDFS/AsynchronousReadBufferFromHDFS.cpp

```diff
@@ -92,12 +100,22 @@ std::optional<size_t> AsynchronousReadBufferFromHDFS::getFileSize()
     return impl->getFileSize();
 }
 
+String AsynchronousReadBufferFromHDFS::getFileName() const
+{
+    return impl->getFileName();
+}
+
 
 bool AsynchronousReadBufferFromHDFS::nextImpl()
 {
     if (!hasPendingDataToRead())
         return false;
 
+    ++next_times;
+    sum_interval += interval_watch.elapsedMicroseconds();
+
+    Stopwatch next_watch;
+    Int64 wait = -1;
```

Comment:
why -1 instead of 0? below there is `sum_wait += wait;` unconditionally, event if `wait` was not updated, so `sum_wait` is ok to be negative?

Reply:
When prefetch is not used, wait is meaningless, so I set its value to -1 .